galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.selenium.manager/src/main/java/dev/galasa/selenium/internal/SeleniumEnvironment.java

```diff
@@ -96,35 +102,37 @@ private String allocateSlot() throws ResourceUnavailableException, SeleniumManag
 		String slotName = "";
 		int currentSlots = 0;
 		try {
-			slots = dss.get(slotKey);
-			if (slots != null) {
-				currentSlots = Integer.valueOf(slots);
-			}
-			
-			if (currentSlots >= SeleniumDriverMaxSlots.get()) {
-				throw new ResourceUnavailableException("Failed to provsion. No slots avilable");
-			}			
-			
-			String slotNamePrefix = "SeleniumSlot_" + this.runName + "_";
-			
-			int counter = 0;
 			while ("".equals(slotName)) {
-				String slotNameAttempt = slotNamePrefix + counter;
+				slots = dss.get(slotKey);
+				if (slots != null) {
+					currentSlots = Integer.valueOf(slots);
+				}
+				
+				if (currentSlots >= SeleniumDriverMaxSlots.get()) {
+					throw new ResourceUnavailableException("Failed to provsion. No slots avilable");
+				}			
+				
+				String slotNamePrefix = "SeleniumSlot_" + this.runName + "_";
+			
+				String slotNameAttempt = slotNamePrefix + sessions;
 				try {
+					currentSlots++;
 					dss.performActions(
 					new DssSwap("driver.slot." + slotNameAttempt, null, runName),
-					new DssSwap(slotKey, slots, String.valueOf(currentSlots+1))
+					new DssSwap(slotKey, slots, String.valueOf(currentSlots))
 					);
 					slotName = slotNameAttempt;
 					break;
 				} catch (DynamicStatusStoreException e) {
-					counter++;
+					logger.warn("Failed to get slot: " + slotNameAttempt + ". Retrying... ");
```

Comment:
why warn???  trace is better

Reply:
only warn if it could possibly affect the result of the test or cleanup.   dont warn as part of the normal process or resolution.

Reply:
Still not happy with this as it could get into an endless loop if there is a problem with etcd.